website/static/js/setPassword.js

```diff
@@ -0,0 +1,267 @@
+'use strict';
+
+var ko = require('knockout');
+require('knockout.validation');
+var $ = require('jquery');
+var zxcvbn = require('zxcvbn');
+
+var $osf = require('./osfHelpers');
+
+
+ko.validation.rules.complexity = {
+    validator: function (val, minimumComplexity) {
+        return zxcvbn(val).score >= minimumComplexity;
+    },
+    message: 'Please enter a more complex password.'
+};
+
+ko.validation.registerExtenders();
+
+
+// Accepts a few different types of arguments, depending on
+// the desired fields and type of password reset.
+// types:
+//    - signup: required fields for Name, email, email confirmation, password, and password strength
+//    - reset: fields for password, strength, and password confirmation (for when you don't know your old password)
+//    - change: change your password when you know your old one
+var ViewModel = function(passwordViewType, submitUrl, campaign, redirectUrl) {
+
+    var self = this;
+
+    self.typedPassword = ko.observable('');
+
+    self.passwordInfo = ko.computed(function() {
```

Comment:
Minor: this can be a `pureComputed`. Usually this doesn't matter much, but I'm not sure how efficient zxcvbn is, so it might be worth it here.